Userland/Libraries/LibCore/Stream.h

```diff
@@ -671,6 +671,54 @@ class BufferedHelper {
         return StringView { TRY(read_until(buffer, "\n"sv)) };
     }
 
+    // Read into the buffer until \n is encountered, resizing the buffer as needed.
+    // Returns the bytes read as a StringView.
+    ErrorOr<StringView> read_line_unbounded(ByteBuffer& buffer)
+    {
+        if (!buffer.is_empty())
+            buffer.clear();
+
+        bool is_line_complete = false;
+        Bytes span_to_copy;
+        Bytes remaining_data;
+
+        while (!(is_line_complete || is_eof())) {
+            // m_buffer is populated by can_read_line()
+            auto is_line_readable = TRY(can_read_line());
+            auto data_in_buffer = m_buffer.span().slice(0, m_buffered_size);
+
+            if (data_in_buffer.is_empty())
+                break;
+
+            Optional<size_t> newline_index;
+
+            if (is_line_readable) {
+                is_line_complete = true;
+                newline_index = data_in_buffer.find_first_index('\n');
```

Comment:
In this code, `is_line_complete == newline_index.has_value()`. When you find a newline index, you can break immediately instead of waiting on the loop condition which can then be simplified to `while (!is_eof())`.